### ISSUE TITLE
Update a record cause duplicate of data

### ISSUE THREAD

**Issue by daoan1412** [objectbox/objectbox-swift#18]
```swift
// objectbox: entity
struct Author {
    let id: Id
    let name: String
    let age: Int
}

let exampleEntityBox = store.box(for: Author.self)
var exampleEntity = Author(id: 0, name: "Nnedi Okorafor", age: 45)

exampleEntity.id = try exampleEntityBox.put(exampleEntity)

// update record
exampleEntity.age = 50
try exampleEntityBox.put(exampleEntity)


exampleEntityBox.subscribe { records, _ in
      //--> duplicate records
}
```

```
Environment:
Xcode 11 GM seed 2
ObjectBox: 1.0.0-rc.10
```

**Comment by uli-objectbox** [comment]
This works for me. How are you creating your store, and have you checked that the object wasn't already in the store from a previous run?

I had to modify your code slightly to make it compile. `id` is a `let` constant, so the compiler will complain about the assignment to it, and you're ignoring the return value from `subscribe()`.

Once I changed `id` to a `var` and added a `_ =` before the subscribe (in a real program, you'd of course keep the subscription around in a property or so), the code worked, and the records array contains only the single object I wrote.
